lib/bundles/inspec-compliance/api.rb

```diff
@@ -18,12 +19,15 @@ class API
     # return all compliance profiles available for the user
     # the user is either specified in the options hash or by default
     # the username of the account is used that is logged in
-    def self.profiles(config)
+    def self.profiles(config) # rubocop:disable PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/AbcSize, Metrics/MethodLength
```

Comment:
You're getting dinged for these things because it's past time to treat API as an abstract parent class and make concrete subclasses for the differing targets; `API_Compliance`, `API_Automate`, and `API_Automate2`. Almost all of the conditional code you had to expand here goes away.

Reply:
Yeah, I really wanted to break up a lot of this code into sub classes like this. I will take a shot at it and see if I can fit in into the timeline.

Reply:
Started down this path a bit tonight, its much cleaner but there is a LOT to do here. We call the compliance stuff all over and will need to be redone. I will create a ticket to get this refactor in as its needed but out of scope for now.

Reply:
https://github.com/chef/inspec/issues/2964 added, thanks for the feedback.